pyworklistserver/worklist_server.py

```diff
@@ -44,11 +70,28 @@ def stop(self):
 
     def _on_find(self, event, app_logger):
         """ Event handler for C-FIND requests """
-        worklist_items = self._worklist_factory.get_worklist()
-        app_logger.info('Created worklist with {} exams'.format(len(worklist_items)))
-        for worklist_item in worklist_items:
+        seed = self.get_seedNumber()
+        random.seed(seed)
+        self.log_seed(seed)
+
+        totalRate = self._config_values.rateOfRandomExams + self._config_values.rateOfCleanExams
+        totalExams = random.randrange(self._config_values.minAmountOfWorklistExams, self._config_values.maxAmountOfWorklistExams)
+
+        for i in range (totalExams):

```

Comment:
The _on_find function should be as simple as possible, and have as few responsibilities as possible. Therefore, please split out a new worklist generator class that returns the worklist that the _on_find should return. This new class can have a clean worklist generator and a random worklist generator internally, to provide the type of worklist you want.

Reply:
fixed in issue #10